apps/lead/tests/test_mutations.py

```diff
@@ -389,6 +389,45 @@ def _query_check(**kwargs):
         self.assertMatchSnapshot(response, 'success')
         self.assertEqual(lead_count + 2, Lead.objects.count())
 
+    def test_lead_bulk_delete_validation(self):
+        query = '''
+            mutation MyMutation ($projectId: ID! $input: [ID!]) {
+              project(id: $projectId) {
+                leadBulk(deleteIds: $input) {
+                    errors
+                    deletedResult {
+                        title
+                    }
+                }
+              }
+            }
+        '''
+
+        project = ProjectFactory.create()
+        user = UserFactory.create()
+        project.add_member(user, role=self.project_role_member)
+
+        lead1 = LeadFactory.create(project=project)
+        lead2 = LeadFactory.create(project=project)
+
+        lead_count = Lead.objects.count()
+
+        minput = [str(lead1.pk), str(lead2.pk)]
+
+        def _query_check(**kwargs):
+            return self.query_check(query, minput=minput, variables={'projectId': project.pk}, **kwargs)
+
+        # Error without login
+        _query_check(assert_for_error=True)
+
+        # --- login
+        self.force_login(user)
+        self.assertEqual(lead_count, Lead.objects.count())
+        # Success with normal lead (with project membership)
+        result = _query_check()['data']['project']['leadBulk']
+        self.assertMatchSnapshot(result, 'success')
+        self.assertEqual(lead_count - 2, Lead.objects.count())
```

Comment:
Make sure to add test for different user with different permission.
Also Instead of count, let's assert for id.